u8timeseries/models/fft.py

```diff
@@ -0,0 +1,201 @@
+    """
+    return all(map(lambda attr: getattr(ts_1, attr) == getattr(ts_2, attr), required_matches))
+
+def crop_to_match_seasons(series: TimeSeries, required_matches: Optional[set]) -> TimeSeries:
+    """
+    Crops a given TimeSeries 'series' that will be used as a training set in such
+    a way that its first entry has a timestamp that matches the first timestamp
+    right after the end of 'series' in all attributes given in 'required_matches'.
+    If no such timestamp can be found, the original TimeSeries instance is returned.
+    If the value of 'required_matches' is 'None', the original TimeSeries instance is returned.
+
+    :param series: TimeSeries instance to be cropped.
+    :param ts_2: Second timestamp that will be compared.
+    :required_matches: A set of pd.Timestamp attributes which will be used to choose the cropping point.
+    :return: New TimeSeries instance that is cropped as described above.
+    """
+    if (required_matches is None): return series
+
+    first_ts = series._series.index[0]
+    freq = first_ts.freq
+    pred_ts = series._series.index[-1] + freq
+
+    # start at first timestamp of given series and move forward until a matching timestamp is found
+    curr_ts = first_ts
+    while (curr_ts < pred_ts - 4 * freq):
+        curr_ts += freq
+        if compare_timestamps_on_attributes(pred_ts, curr_ts, required_matches):
+            new_series = series.drop_before(curr_ts)
+            return new_series
+    
+    logger.warning("No matching timestamp could be found, returning original TimeSeries.")
+    return series
+
+class FFT(AutoRegressiveModel):
+
+    def __init__(self, nr_freqs_to_keep: Optional[int] = None, required_matches: Optional[set] = None, 
+                 trend: bool = None, trend_poly_degree: int = 3, automatic_matching=False):
+        """
+        Forecasting based on a discrete fourier transform using FFT of the (cropped and detrended) training sequence
+        with subsequent selection of the most significant frequencies to remove noise from the prediction.
+
+        :param nr_freqs_to_keep: The total number of frequencies that will be used for forecasting.
+        :param required_matches: The attributes of pd.Timestamp that will be used to create a training
+                                 sequence that is cropped at the beginning such that the first timestamp 
+                                 of the training sequence and the first prediction point have matching 'phases'.
+                                 If the series has a yearly seasonality, include 'month', if it has a monthly 
+                                 seasonality, include 'day', etc.
+        :param trend: Boolean value indicating whether or not detrending will be applied before performing DFT.
+        :param trend_poly_degree: The degree of the polynomial that will be used for detrending.
+        :param automatic_matching: If set to True, the model tries to find the pd.Timestamp attributes that 
```

Comment:
I think you don't need this one. You can just do the automatic mode when `required_matches` is None.

Reply:
done